src/WebJobs.Extensions.ApiHub/ApiHubFile.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.Azure.ApiHub;
+using Microsoft.Azure.WebJobs.Extensions.ApiHub.Common;
+using System;
+using System.IO;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.WebJobs.Extensions.ApiHub
+{
+    class ApiHubFile : IFileStreamProvider
+    {
+        private readonly IFileItem _fileSource;
+
+        public string Path
+        {
+            get
+            {
+                return _fileSource.Path;
+            }
+        }
+
+        public ApiHubFile(IFolderItem rootFolder, string path)
+        {
+            _fileSource = rootFolder.CreateFileAsync(path, true).GetAwaiter().GetResult();
```

Comment:
+1. We shouldn't call .GetResult().
Instead, try something like 
internal static async Task New(IFolderItem rootFolder, string path)
{
var x =  = await ...
return new ApiHubFile(x)
}